resources/zib-Patient.xml

```diff
@@ -0,0 +1,722 @@
+         <definition value="The contact’s telephone number and/or e-mail address."/>
+         <comment value="The cardinality mismatch between the zib (0..1) and FHIR (0..*) is explained by the containers TelephoneNumbers (0..*) and EmailAddresses (0..*) inside the ContactInformation model. The FHIR datatype ContactPoint does not have these containers."/>
+         <alias value="Contactgegevens"/>
+         <mapping>
+            <identity value="zib-contactperson-v3.4-2020EN"/>
+            <map value="NL-CM:3.1.6"/>
+            <comment value="ContactInformation"/>
+         </mapping>
+         <mapping>
+            <identity value="zib-contactinformation-v1.2-2020EN"/>
+            <map value="NL-CM:20.6.1"/>
+            <comment value="ContactInformation"/>
+         </mapping>
+      </element>
+      <element id="Patient.contact.telecom:telephoneNumbers">
+         <path value="Patient.contact.telecom"/>
+         <sliceName value="telephoneNumbers"/>
+         <type>
+            <code value="ContactPoint"/>
+            <profile value="http://nictiz.nl/fhir/StructureDefinition/zib-ContactInformation-TelephoneNumbers"/>
+         </type>
+      </element>
+      <element id="Patient.contact.telecom:e-mailAddresses">
+         <path value="Patient.contact.telecom"/>
+         <sliceName value="e-mailAddresses"/>
+         <type>
+            <code value="ContactPoint"/>
+            <profile value="http://nictiz.nl/fhir/StructureDefinition/zib-ContactInformation-E-mailAddresses"/>
+         </type>
+      </element>
+      <element id="Patient.contact.address">
+         <path value="Patient.contact.address"/>
+         <short value="AddressInformation"/>
+         <definition value="Contact’s address information."/>
+         <alias value="Adresgegevens"/>
+         <type>
+            <code value="Address"/>
+            <profile value="http://nictiz.nl/fhir/StructureDefinition/zib-AddressInformation"/>
+         </type>
+         <mapping>
+            <identity value="zib-contactperson-v3.4-2020EN"/>
+            <map value="NL-CM:3.1.5"/>
+            <comment value="AddressInformation"/>
+         </mapping>
+      </element>
+      <element id="Patient.communication">
+         <path value="Patient.communication"/>
+         <short value="LanguageProficiency"/>
+         <definition value="Root concept of the LanguageProficiency information model. This concept contains all data elements of the LanguageProficiency information model."/>
+         <alias value="Taalvaardigheid"/>
+         <mapping>
+            <identity value="zib-languageproficiency-v3.2-2020EN"/>
+            <map value="NL-CM:7.12.1"/>
+            <comment value="LanguageProficiency"/>
+         </mapping>
+      </element>
+      <element id="Patient.communication.extension">
+         <path value="Patient.communication.extension"/>
+         <slicing>
+            <discriminator>
+               <type value="value"/>
+               <path value="url"/>

```

Comment:
Ik heb het idee dat dit niet goed gaat. Hoe weet de validator in elke extensie slice die moet zijn?

Hebben we dit gecontroleerd met een voorbeeld?

Ik denk dat er een discriminator bij moet, of dit path aanpassen naar extension('http://hl7.org/fhir/StructureDefinition/patient-proficiency').extension('type').value

Al doe ik dit nu uit mijn hoofd, weet niet zeker of die tweede extension werkt zo.

Reply:
Path aanpassen gaat denk ik niet werken, want er zijn nóg twee extensies die wel op url moeten. Dus dan 2e discriminator toevoegen? Dat kan lijkt me niet vanuit Forge UI?

Zal het even wat uitgebreider testen.

Reply:
Dit is weer teruggedraaid. Dus ik sluit deze 'conversation'.